Calibration/HcalAlCaRecoProducers/src/AlCaHOCalibProducer.cc

```diff
@@ -196,86 +196,52 @@ class AlCaHOCalibProducer : public edm::EDProducer {
 
   float pedestal[netamx][nphimx][ncidmx]; 
 
-  std::string digiLabel;
-  
-  bool debug;
-  std::string theRootFileName;
-
-  //GMA #ifdef DEBUG_OFFLINE_GM
-
-  TH1F* libhoped;
-  TH1F* libhoped1;
-
-  TH1F* allhotime;
-  TH1F* hotime[ntrgp_gm+1];
-  TH1F* hopedtime;
 
-  TProfile* hopedpr;  
-  TH1F* hopedrms;  
-  TH1F* hst_hopedrms;    
-
-  TProfile* hopeak[ntrgp_gm+1];
-  TProfile* horatio;
-
-  TH1F* Nallhotime;
-  TH1F* Nhotime[ntrgp_gm+1];
-  TH1F* Nhopedtime;
-
-  TH1F* allhb1;
-  TH1F* allhb2;
-  TH1F* allhb3;
-
-  TH1F* Nallhb1;
-  TH1F* Nallhb2;
-  TH1F* Nallhb3;
-
-  TProfile* hb1pedpr;  
-  TH1F* hb1pedrms;  
-  TH1F* hst_hb1pedrms;    
+  /* 
+     std::string digiLabel;
+     bool debug;
+  */
+  std::string theRootFileName;
 
-  TH1F* ho_occupency[5];  
+  TH2F* ho_occupency[5];  
 
-  bool m_hotime;
-  //GM #endif
+  bool m_occupancy=false; //true;
 
   edm::InputTag muonTags_;   // cosmicMuons or standAloneMuons
 
+#ifdef COSMIC
   edm::EDGetTokenT<reco::TrackCollection> tok_muons_;
+#else
+  edm::EDGetTokenT<edm::View<reco::Muon> > tok_muons_;
+  edm::EDGetTokenT<reco::VertexCollection> tok_vertex_;
+  //  edm::EDGetTokenT<LumiDetails> tok_lumi_;
+  edm::EDGetTokenT<LumiScalersCollection> tok_lumi_;
+
+#endif
```

Comment:
It is quite unclear to me how you decide if using Cosmic or collision data. In other words: I fully understand that you want to consume different collections, but it is not clear which modules defines `COSMIC` in the pre-processor in case you are looking at cosmic data.